texar/modules/encoders/gpt2_encoder.py

```diff
@@ -87,7 +92,7 @@ def __init__(self,
             hparams=self._hparams.decoder)
 
         if self.pretrained_model_dir:
-            gpt2_utils.init_gpt2_checkpoint(self, self.pretrained_model_dir)
+            init_gpt2_checkpoint(self, self.pretrained_model_dir)
```

Comment:
If the first argument to `init_*_checkpoint` is `self`, why not just make it a method of `PretrainedBase`? This way we can call `init_checkpoint` in the base class constructor.